### PR TITLE
[Hinty] Typing of route & route6

### PR DESCRIPTION
**Pull request by gpotter2**
- adds typing to `scapy.route` and `scapy.route6`
- **requires https://github.com/secdev/scapy/pull/2859 (utils.py typing)**

(note: I will add support for `utils6` much much later because unlike `utils` it depends on a lot of other files...)

### CODE REVIEW

File: scapy/route.py [hunk 1]
```diff
@@ -46,13 +62,20 @@ def __repr__(self):
         return pretty_list(rtlst,
                            [("Network", "Netmask", "Gateway", "Iface", "Output IP", "Metric")])  # noqa: E501
 
-    def make_route(self, host=None, net=None, gw=None, dev=None, metric=1):
+    def make_route(self,
+                   host=None,  # type: Optional[str]
+                   net=None,  # type: Optional[str]
+                   gw=None,  # type: Optional[str]
+                   dev=None,  # type: Optional[Any]
```
Comment (guedou) [inline scapy/route.py:69]: Can't we do better than Any here?

File: scapy/route6.py [hunk 2]
```diff
@@ -288,12 +317,12 @@ def route(self, dst=None, dev=None, verbose=conf.verb):
         best_plen = (paths[0][0], paths[0][1])
         paths = [x for x in paths if (x[0], x[1]) == best_plen]
 
-        res = []
-        for p in paths:  # Here we select best source address for every route
-            tmp = p[2]
-            srcaddr = get_source_addr_from_candidate_set(dst, tmp[1])
+        res = []  # type: List[Tuple[int, int, Tuple[str, str, str]]]
+        for path in paths:  # we select best source address for every route
+            tup = path[2]
```
Comment (guedou) [inline scapy/route6.py:322]: tmp is better than tup IMHO.

**Review by guedou — DISMISSED** [review]
I made two simple comments. Otherwise this looks really good.

File: scapy/route6.py (hunk 2 above)
Comment (gpotter2) [reply scapy/route6.py:322]: For sure but `tmp` is used a bit later with a different type :/

Comment (guedou) [reply scapy/route6.py:322]: I understand. tmp_candidates or candidates will be perfect.

Comment (gpotter2) [reply scapy/route6.py]: I chose `tmp_c` so that it doesn't exceed the max line lengthS

**Comment by gpotter2** [comment]
All done !

**Pull request merged**